src/metrics.py

```diff
@@ -2,26 +2,84 @@
 #  See LICENSE file for licensing details.
 
 """Models and functions for the metric events."""
+import json
+import logging
+import time
 
-from pydantic import BaseModel, conint
+import requests
+from pydantic import BaseModel, NonNegativeInt
+
+import promtail
+
+PROMTAIL_PUSH_API_URL = "http://localhost:3100/loki/api/v1/push"
+
+requests_session = None  # pylint: disable=invalid-name
+logger = logging.getLogger(__name__)
 
 
 class Event(BaseModel):
-    """Base class for metric events."""
+    """Base class for metric events.
 
-    timestamp: conint(ge=0)
-    """The UNIX timestamp of the event."""
+    Attrs:
+         timestamp: The UNIX time stamp of the time at which the event was originally issued.
+    """
+
+    timestamp: NonNegativeInt
 
 
 class RunnerInstalled(Event):
-    """Metric event for when a runner is installed."""
+    """Metric event for when a runner is installed.
+
+    Attrs:
+        flavor: Describes the characteristics of the runner.
+          The flavour could be for example "small".
+        duration: The duration of the installation in seconds.
+    """
 
-    name: str = "runner_installed"
-    """The name of the event."""
     flavor: str
-    """The flavor of the runner."""
-    duration: conint(ge=0)
-    """The duration of the installation in seconds."""
+    duration: NonNegativeInt
+
+
+def _get_session() -> requests.Session:
+    """Get a requests session.
+
+    Returns:
+        A requests session.
+    """
+    # We try to avoid capsuling the session in an object to be passed to issue_event,
+    # therefore we use the singleton pattern with a global variable.
+    global requests_session  # pylint: disable=global-statement
```

Comment:
Are there any alternatives here? Needing to maintain a session can be a reason to move to a class rather than a collection of functions. Global variables are usually not recommended

Reply:
I have removed the use of session here. Its not required, it could give some performance improvement but most likely not really felt, and I believe its not worth it if it increases implementation complexity (e.g. moving to a class or using a singleton/factory pattern to retrieve the session).